diff -Naur mac-sdk-4.42/Makefile mac-sdk-4.42-gcc/Makefile
--- mac-sdk-4.42/Makefile	1970-01-01 00:00:00 +0000
+++ mac-sdk-4.42-gcc/Makefile	2018-11-15 04:41:58 +0000
@@ -0,0 +1,103 @@
+# Find system
+UNAME	 = $(shell uname)
+
+VERSION	  = 4
+
+CXXOPTS	  = -I Source/Shared -I Source/MACLib -c
+LDOPTS	  = -lstdc++
+
+DLLLDOPTS = -shared
+CMDLDOPTS = 
+
+ifneq ($(findstring MINGW,$(UNAME)),)
+	SHAREDEXT  = dll
+
+	DLLNAME	   = Source/MACDll.$(SHAREDEXT)
+	LIBNAME	   = Source/libMAC.a
+	CMDNAME	   = Source/MAC.exe
+
+	CXXOPTS	  += -D PLATFORM_WINDOWS
+
+	DLLLDOPTS += -Wl,--kill-at,--out-implib,$(LIBNAME)
+	CMDLDOPTS += -municode -mconsole -Wl,--enable-stdcall-fixup
+else
+	SHAREDEXT  = so
+
+	DLLNAME	   = libMAC.$(SHAREDEXT).$(VERSION)
+	CMDNAME	   = mac
+
+	CXXOPTS	  += -fPIC
+
+#	Use the .dylib extension for shared objects on macOS.
+	ifeq ($(UNAME),Darwin)
+		SHAREDEXT = dylib
+
+		DLLNAME	  = libMAC.$(VERSION).$(SHAREDEXT)
+	endif
+
+#	Set folders if not provided.
+	ifndef prefix
+		prefix = /usr/local
+	endif
+
+	ifndef bindir
+		bindir = $(prefix)/bin
+	endif
+
+	ifndef libdir
+		libdir = $(prefix)/lib
+	endif
+
+	ifndef includedir
+		includedir = $(prefix)/include
+	endif
+
+#	Define PLATFORM_LINUX on all non-Apple platforms.
+	ifeq ($(UNAME),Darwin)
+		CXXOPTS += -D PLATFORM_APPLE
+	else
+		CXXOPTS += -D PLATFORM_LINUX
+	endif
+endif
+
+SHAREDOBJECTS = Source/Shared/CharacterHelper.o Source/Shared/CircleBuffer.o Source/Shared/GlobalFunctions.o Source/Shared/StdLibFileIO.o Source/Shared/WinFileIO.o
+LIBOBJECTS    = Source/MACLib/APECompress.o Source/MACLib/APECompressCore.o Source/MACLib/APECompressCreate.o Source/MACLib/APEDecompress.o Source/MACLib/APEHeader.o Source/MACLib/APEInfo.o Source/MACLib/APELink.o Source/MACLib/APESimple.o Source/MACLib/APETag.o Source/MACLib/BitArray.o Source/MACLib/MACLib.o Source/MACLib/MACProgressHelper.o Source/MACLib/MD5.o Source/MACLib/NewPredictor.o Source/MACLib/NNFilter.o Source/MACLib/Prepare.o Source/MACLib/UnBitArray.o Source/MACLib/UnBitArrayBase.o Source/MACLib/WAVInputSource.o Source/MACLib/Old/Anti-Predictor.o Source/MACLib/Old/AntiPredictorExtraHigh.o Source/MACLib/Old/AntiPredictorFast.o Source/MACLib/Old/AntiPredictorHigh.o Source/MACLib/Old/AntiPredictorNormal.o Source/MACLib/Old/APEDecompressCore.o Source/MACLib/Old/APEDecompressOld.o Source/MACLib/Old/UnBitArrayOld.o Source/MACLib/Old/UnMAC.o
+DLLOBJECTS    = Source/MACDll/MACDll.o
+CMDOBJECTS    = Source/Console/Console.o
+
+all: $(DLLNAME) $(CMDNAME)
+
+clean:
+	rm -f $(DLLNAME) $(LIBNAME) $(SHAREDOBJECTS) $(LIBOBJECTS) $(DLLOBJECTS) $(CMDNAME) $(CMDOBJECTS)
+
+install:
+	mkdir -p $(bindir)
+	cp $(CMDNAME) $(bindir)
+
+	mkdir -p $(libdir)
+	cp $(DLLNAME) $(libdir)
+	ln -sf $(DLLNAME) $(libdir)/libMAC.$(SHAREDEXT)
+
+	mkdir -p $(includedir)/MAC
+	cp Shared/*.h $(includedir)/MAC
+
+$(DLLNAME): $(SHAREDOBJECTS) $(LIBOBJECTS) $(DLLOBJECTS)
+	$(CC) $(SHAREDOBJECTS) $(LIBOBJECTS) $(DLLOBJECTS) $(LDOPTS) $(LDFLAGS) $(DLLLDOPTS) -o $@
+
+$(CMDNAME): $(DLLNAME) $(CMDOBJECTS)
+	$(CC) $(SHAREDOBJECTS) $(CMDOBJECTS) $(DLLNAME) $(LDOPTS) $(LDFLAGS) $(CMDLDOPTS) -o $@
+
+Source/Shared/%.o: Source/Shared/%.cpp
+	$(CXX) $(CXXOPTS) $(CXXFLAGS) $< -o $@
+
+Source/MACLib/%.o: Source/MACLib/%.cpp
+	$(CXX) $(CXXOPTS) $(CXXFLAGS) $< -o $@
+
+Source/MACLib/Old/%.o: Source/MACLib/Old/%.cpp
+	$(CXX) $(CXXOPTS) $(CXXFLAGS) $< -o $@
+
+Source/MACDll/%.o: Source/MACDll/%.cpp
+	$(CXX) $(CXXOPTS) $(CXXFLAGS) $< -o $@
+
+Source/Console/%.o: Source/Console/%.cpp
+	$(CXX) $(CXXOPTS) $(CXXFLAGS) $< -o $@
diff -Naur mac-sdk-4.42/Shared/All.h mac-sdk-4.42-gcc/Shared/All.h
--- mac-sdk-4.42/Shared/All.h	2018-12-09 20:33:25 +0000
+++ mac-sdk-4.42-gcc/Shared/All.h	2018-12-19 19:34:27 +0000
@@ -87,7 +87,7 @@
 	#define PLATFORM_x64
 #endif
 
-#if __x86_64__
+#if __x86_64__ || __aarch64__
     #define PLATFORM_x64
 #endif
 
diff -Naur mac-sdk-4.42/Source/Console/Console.cpp mac-sdk-4.42-gcc/Source/Console/Console.cpp
--- mac-sdk-4.42/Source/Console/Console.cpp	2018-12-09 19:12:19 +0000
+++ mac-sdk-4.42-gcc/Source/Console/Console.cpp	2018-12-16 10:32:35 +0000
@@ -21,13 +21,18 @@
 #define UNDEFINED_MODE		-1
 
 // use 8 bit character functions on non-Windows platforms
-#ifndef PLATFORM_WINDOWS
-	#define _tmain main
-	#define _tcscpy strcpy
-	#define _tcsncpy strncpy
-	#define _tcsnicmp strncasecmp
+#ifdef PLATFORM_WINDOWS
+	#ifndef _MSC_VER
+		#define _tcscpy_s(dst, num, src) wcscpy(dst, src)
+		#define _tcsncpy_s(dst, num, src, count) wcsncpy(dst, src, count)
+	#endif
+#else
+	#define _tmain(argc, argv) main(argc, argv)
+	#define _tcscpy_s(dst, num, src) strcpy(dst, src)
+	#define _tcsncpy_s(dst, num, src, count) strncpy(dst, src, count)
+	#define _tcsnicmp(str1, str2, count) strncasecmp(str1, str2, count)
 	#define _ftprintf fprintf
-	#define _ttoi atoi
+	#define _ttoi(str) atoi(str)
 	#define TCHAR char
 	#define _T(x) x
 #endif
@@ -77,13 +82,14 @@
 	_ftprintf(stderr, _T("Progress: %.1f%% (%.1f seconds remaining, %.1f seconds total)          \r"), 
 		dProgress * 100, dRemaining, dElapsed);
 
-	// don't forget to flush!
+	// don't forget to flush!
 	fflush(stderr);
 }
 
 /***************************************************************************************
 CtrlHandler callback
 ***************************************************************************************/
+#ifdef PLATFORM_WINDOWS
 static BOOL CALLBACK CtrlHandlerCallback(DWORD dwCtrlTyp)
 {
 	switch (dwCtrlTyp)
@@ -102,6 +108,7 @@
 	ExitProcess(666);
 	return TRUE;
 }
+#endif
 
 /***************************************************************************************
 Main (the main function)
diff -Naur mac-sdk-4.42/Source/MACLib/UnBitArray.cpp mac-sdk-4.42-gcc/Source/MACLib/UnBitArray.cpp
--- mac-sdk-4.42/Source/MACLib/UnBitArray.cpp	2018-11-06 01:51:22 +0000
+++ mac-sdk-4.42-gcc/Source/MACLib/UnBitArray.cpp	2018-12-15 22:39:30 +0000
@@ -61,6 +61,7 @@
 
 inline uint32 CUnBitArray::DecodeByte()
 {
+    if ((m_nCurrentBitIndex + 8) >= (m_nGoodBytes * 8))
 	EnsureBitsAvailable(8, true);
 
     // read byte
diff -Naur mac-sdk-4.42/Source/Shared/All.h mac-sdk-4.42-gcc/Source/Shared/All.h
--- mac-sdk-4.42/Source/Shared/All.h	2018-12-09 20:33:25 +0000
+++ mac-sdk-4.42-gcc/Source/Shared/All.h	2018-12-19 19:34:46 +0000
@@ -87,7 +87,7 @@
 	#define PLATFORM_x64
 #endif
 
-#if __x86_64__
+#if __x86_64__ || __aarch64__
     #define PLATFORM_x64
 #endif
 
@@ -162,6 +162,12 @@
             #define ASSERT(e)                            
         #endif
     #endif
+    #if !defined _MSC_VER
+        #define wcsncpy_s(A, B, COUNT) wcsncpy(A, B, COUNT)
+        #define sprintf_s(A, B, C, ...) sprintf(A, C, __VA_ARGS__)
+        #define _stprintf_s(A, B, C, ...) _stprintf(A, C, __VA_ARGS__)
+        #define wcscpy_s(A, B, C) wcscpy(A, C)
+    #endif
 #else
     #define IO_USE_STD_LIB_FILE_IO
     #define IO_HEADER_FILE                              "StdLibFileIO.h"
@@ -175,9 +181,9 @@
     #define TICK_COUNT_READ(VARIABLE)                   { struct timeval t; gettimeofday(&t, NULL); VARIABLE = t.tv_sec * 1000000LLU + t.tv_usec; }
     #define TICK_COUNT_FREQ                             1000000
     #define ASSERT(e)
-    #define wcsncpy_s(A, B, COUNT) wcscpy(A, B)
+    #define wcsncpy_s(A, B, COUNT) wcsncpy(A, B, COUNT)
     #define wcscpy_s(A, B, C) wcscpy(A, C)
-    #define sprintf_s(A, B, C, D) sprintf(A, C, D)
+    #define sprintf_s(A, B, C, ...) sprintf(A, C, __VA_ARGS__)
 #endif
 
 /*****************************************************************************************
diff -Naur mac-sdk-4.42/Source/Shared/WinFileIO.cpp mac-sdk-4.42-gcc/Source/Shared/WinFileIO.cpp
--- mac-sdk-4.42/Source/Shared/WinFileIO.cpp	2018-12-07 12:49:56 +0000
+++ mac-sdk-4.42-gcc/Source/Shared/WinFileIO.cpp	2018-12-15 22:40:50 +0000
@@ -55,7 +55,7 @@
         m_bReadOnly = false;
     }
     
-    wcscpy_s(m_cFileName, pName);
+    wcscpy(m_cFileName, pName);
 
     return 0;
 }
@@ -132,7 +132,7 @@
 
 int CWinFileIO::GetName(wchar_t * pBuffer)
 {
-    wcscpy_s(pBuffer, MAX_PATH, m_cFileName);
+    wcscpy(pBuffer, m_cFileName);
     return 0;
 }
 
@@ -154,7 +154,7 @@
 
     m_bReadOnly = false;
     
-    wcscpy_s(m_cFileName, pName);
+    wcscpy(m_cFileName, pName);
 
     return 0;
 }
diff -Naur mac-sdk-4.42/configure mac-sdk-4.42-gcc/configure
--- mac-sdk-4.42/configure	1970-01-01 00:00:00 +0000
+++ mac-sdk-4.42-gcc/configure	2018-06-13 20:39:56 +0000
@@ -0,0 +1 @@
+#!/bin/sh
