--- a/runtime/application/componentspecs.cpp	2018-11-25 14:16:45.000000000 +0100
+++ b/runtime/application/componentspecs.cpp	2018-12-17 21:48:03.414597108 +0100
@@ -309,7 +309,7 @@
 #elif defined __NetBSD__
 	static const char	*places[] = { "%APPDIR/codecs/cmdline/%COMMAND", "/usr/bin/%COMMAND", "/usr/local/bin/%COMMAND", "/usr/pkg/bin/%COMMAND", NIL };
 #else
-	static const char	*places[] = { "%APPDIR/codecs/cmdline/%COMMAND", "/usr/bin/%COMMAND", "/usr/local/bin/%COMMAND", NIL };
+	static const char	*places[] = { "%APPDIR/../usr/bin/%COMMAND", "%APPDIR/codecs/cmdline/%COMMAND", "/usr/bin/%COMMAND", "/usr/local/bin/%COMMAND", NIL };
 #endif
 
 	XML::Document	*document = new XML::Document();
--- a/runtime/common/config.cpp	2018-11-29 00:08:39.000000000 +0100
+++ b/runtime/common/config.cpp	2018-12-17 21:46:53.115465101 +0100
@@ -304,7 +304,7 @@
 		if (configDir != NIL)
 		{
 #if !defined(__WIN32__) && !defined(__HAIKU__)
-			configDir.Append(".");
+			if (!configDir.EndsWith("/.config/")) configDir.Append(".");
 #endif
 			configDir.Append(BoCA::GetApplicationPrefix()).Append(Directory::GetDirectoryDelimiter());
 		}
